### ISSUE TITLE
building the spring boot microservice via dev spaces terminal fails

### ISSUE THREAD

**Issue by hpgrahsl** [redhat-scholars/cloud-native-apps#1]
Issue is in step section 3 of the lab https://redhat-scholars.github.io/cloud-native-apps/cloud-native-apps/03-deploy.html

1. the "proper terminal" named `vscode-*` doesn't seem to be available anymore
2. when trying to fall back to the `universal-developer-image` the `mvn` build always fails most likely due to cpu/mem constraints or similar quota related aspects  ![Screenshot 2022-09-07 at 20 31 05](https://user-images.githubusercontent.com/7030308/188952631-9db75935-a268-4335-b994-d19a275ee619.png)
3. ran `mvn package` multiple times and eventually at least all dependencies where resolved but compilation never worked without the process getting killed (OOM killer?)

only quick workaround is to e.g. build the project locally and work with the resulting jar file in the next step of the lab. root cause still needs to be identified and fixed properly

**Comment by hpgrahsl** [comment]
After playing around a bit with settings in the terminal I'm pretty sure that the problem is just memory related. The maven build needs "too much memory" or at least more than what is available to the container running the terminal. 

The following limits are there and 128M for that is used to run the terminal container isn't sufficient, see below:

<img width="457" alt="Screenshot 2022-09-07 at 21 45 51" src="https://user-images.githubusercontent.com/7030308/188964308-4e19ab71-df32-45c2-ba38-a2bbe2d0760c.png">

**Comment by evanshortiss** [comment]
I have tested this and encountered the same @hpgrahsl. I fixed it by modifying the devfile to increase memory for the *universal-developer-image* as shown:

![Screenshot 2022-09-07 at 2 28 00 PM](https://user-images.githubusercontent.com/1303687/188984748-3e6e9dbb-43de-4262-ba3a-986bf8b91b28.png)

Seems we have two options:

1. Add this step to the instructions
2. Add a devfile to the repo (I will test this shortly)
